commcare_connect/opportunity/helpers.py

```diff
@@ -1,42 +1,65 @@
 from django.db.models import Case, Count, F, Max, Min, Q, Sum, Value, When
 
-from commcare_connect.opportunity.models import CompletedWorkStatus, Opportunity, OpportunityAccess
+from commcare_connect.opportunity.models import (
+    CompletedWorkStatus,
+    Opportunity,
+    OpportunityAccess,
+    UserInvite,
+    VisitValidationStatus,
+)
 
 
 def get_annotated_opportunity_access(opportunity: Opportunity):
     learn_modules_count = opportunity.learn_app.learn_modules.count()
     access_objects = (
-        OpportunityAccess.objects.filter(opportunity=opportunity)
-        .select_related("user", "opportunityclaim")
+        UserInvite.objects.filter(opportunity=opportunity)
```

Comment:
Since one to one fields should be able to traverse the foreign key in either direction, it would be nice to leave this as based on the opportunity access, which would also make the diff much smaller and make the PR less risky

Reply:
The User Invites can exist for users who don't have a valid connect-id. Also in cases where users don't have a connect-id account, we do not create the opportunity access that would lead to no invite status showing for them in the UserStatusTable if the model that supports it is kept to be OpportunityAccess.